src/include/duckdb/storage/table/row_group.hpp

```diff
@@ -80,19 +80,23 @@ class RowGroup : public SegmentBase<RowGroup> {
 	//! The RowGroupCollection this row-group is a part of
 	reference<RowGroupCollection> collection;
 	//! The version info of the row_group (inserted and deleted tuple info)
-	atomic<optional_ptr<RowVersionManager>> version_info;
+	mutable atomic<optional_ptr<RowVersionManager>> version_info;
 	//! The owned version info of the row_group (inserted and deleted tuple info)
-	shared_ptr<RowVersionManager> owned_version_info;
+	mutable shared_ptr<RowVersionManager> owned_version_info;
 	//! The column data of the row_group
-	vector<shared_ptr<ColumnData>> columns;
+	mutable vector<shared_ptr<ColumnData>> columns;
```

Comment:
Can avoid these being `mutable`? For the atomic, if we explicitly use `load()`, I think it does not have to be `mutable`. Maybe we can do a similar trick with the other ones?

Reply:
This is also lazily loaded, in `RowGroup::GetColumn`
```c++
	this->columns[c] =
	    ColumnData::Deserialize(GetBlockManager(), GetTableInfo(), c, start, column_data_reader, types[c]);
```

Reply:
```c++
void RowGroup::SetVersionInfo(shared_ptr<RowVersionManager> version) const {
	owned_version_info = std::move(version);
	version_info = owned_version_info.get();
}

shared_ptr<RowVersionManager> RowGroup::GetOrCreateVersionInfoInternal() {
	// version info does not exist - need to create it
	lock_guard<mutex> lock(row_group_lock);
	if (!owned_version_info) {
		auto new_info = make_shared_ptr<RowVersionManager>(start);
		SetVersionInfo(std::move(new_info));
	}
	return owned_version_info;
}
```
The `version_info` is also lazily loaded